chainerx_cc/chainerx/cuda/cuda_device.cc

```diff
@@ -12,6 +14,50 @@
 
 namespace chainerx {
 namespace cuda {
+namespace cuda_internal {
+
+MemoryKeeper::~MemoryKeeper() {
+    while (!queue_.empty()) {
+        const std::pair<cudaEvent_t, std::shared_ptr<void>>& pair = queue_.front();
+        cudaEventDestroy(pair.first);
+        queue_.pop();
+    }
+}
+
+void MemoryKeeper::Add(cudaStream_t stream, std::shared_ptr<void> memory) {
```

Comment:
This is not a strong opinion but how about removing the stream argument and hard-code it closer to the cuda call avoiding arguments that are "not used".

Reply:
As we should surely implement it, I think it's ok to reserve the parameter.